src/stream/src/executor/sink.rs

```diff
@@ -22,14 +22,13 @@ use itertools::Itertools;
 use risingwave_common::array::stream_chunk::StreamChunkMut;
 use risingwave_common::array::{merge_chunk_row, Op, StreamChunk, StreamChunkCompactor};
 use risingwave_common::catalog::{ColumnCatalog, Field, Schema};
-use risingwave_connector::dispatch_sink;
 use risingwave_connector::sink::catalog::{SinkId, SinkType};
 use risingwave_connector::sink::log_store::{
     LogReader, LogReaderExt, LogStoreFactory, LogWriter, LogWriterExt,
 };
-use risingwave_connector::sink::{
-    build_sink, LogSinker, Sink, SinkImpl, SinkParam, SinkWriterParam,
-};
+use risingwave_connector::sink::{LogSinker, Sink, SinkParam, SinkWriterParam};
+use risingwave_sink_impl::dispatch_sink;
+use risingwave_sink_impl::sink::{build_sink, SinkImpl};
```

Comment:
I'm wondering can we also use the similar tricks for `stream` and `frontend`? 🤔️ If not, interested to know why.

Reply:
It's doable for frontend. I have removed frontend's dependency for frontend.

For stream it's currently not doable because in the trait of `LogSinker` it currently takes a generic parameter `LogReader`, which is not fixed type to form a function pointer. 

To enable this trick for stream, we can have two approaches. First, we can introduce boxed log reader and do dynamic dispatch on the log reader parameter, but the box log reader will create a box future when its method gets called and involves an extra memory allocation and may have performance issue. Second, we can first separate the implementation of log store from the stream crate to a new crate, and wrap all implementation of log reader with an enum, and the connector crate declares a function pointer with such log reader enum as input parameter.

The first method may have performance issue. The second method involves much more work. We can probably investigate the potential in future PR.